core/src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/slick/CompactEntityComponent.scala

```diff
@@ -202,6 +215,128 @@ class CompactEntityQuery(driverComponent: DriverComponent) extends RawSqlQuery w
       WHERE workspace_id = $workspaceId
       GROUP BY entity_type;""".as[EntityTypeAndCount]
 
+  def countEntities(workspaceId: UUID, entityType: String): ReadWriteAction[Int] =
+    concatSqlActions(
+      sql"select count(*) ",
+      fromActiveEntitiesOfTypeInWorkspace(workspaceId, entityType)
+    ).as[Int].map(_.head)
+
+  def countEntitiesWithColumnFilter(workspaceId: UUID,
+                                    entityType: String,
+                                    columnFilter: EntityColumnFilter
+  ): ReadWriteAction[Int] =
+    concatSqlActions(
+      sql"select count(*) ",
+      fromActiveEntitiesOfTypeInWorkspace(workspaceId, entityType),
+      columnFilterCondition(columnFilter)
+    ).as[Int].map(_.head)
+
+  def countEntitiesWithFilterTerms(workspaceId: UUID,
+                                   entityType: String,
+                                   entityQuery: EntityQuery
+  ): ReadWriteAction[Int] =
+    concatSqlActions(
+      sql"select count(*) ",
+      fromActiveEntitiesOfTypeInWorkspace(workspaceId, entityType),
+      filterTermsCondition(entityQuery)
+    ).as[Int].map(_.head)
+
+  def queryEntitiesWithFilterTerms(workspaceId: UUID,
+                                   entityType: String,
+                                   entityQuery: EntityQuery
+  ): SqlStreamingAction[Seq[CompactEntityRecord], CompactEntityRecord, Read] =
+    concatSqlActions(
+      selectCompactEntityColumns,
+      orderByColumns(entityQuery),
```

Comment:
is there any reason to return the `orderByColumns` as part of the select list? The `orderBy(entityQuery)` works fine without them (you'd have to remove the trailing comma from `selectCompactEntityColumns`)